applications/nrf_desktop/doc/porting_guide.rst

```diff
@@ -1,10 +1,152 @@
+      * The application is configured to act as both mouse and keyboard.
+      * Bluetooth uses Nordic's proprietary link layer and is configured to act as a central.
+        Input data come from Bluetooth and are retransmitted to the USB.
+      * |preconfigured_build_types|
+
+.. _porting_guide_adding_board:
+
+Adding a new board
+******************
+
+When adding a new board for the first time, focus on a single configuration.
+Moreover, keep the default ``ZDebug`` build type that the application is built with, and do not add any additional build type parameters.
+
+.. note::
+    The following procedure will use the gaming mouse configuration as example.
+
+To use the nRF Desktop project with your custom board:
+
+1. Define the board by copying the nRF Desktop board files that are the closest match for your hardware.
+   For example, for gaming mouse use ``nrf/boards/arm/nrf52840_pca20041``.
+#. Edit the DTS files to make sure they match the hardware configuration.
+   Pay attention to pins that are used and to the bus configuration for optical sensor.
+#. Copy the project files for the device that is the closest match for your hardware.
+   For example, for gaming mouse these are located at ``applications/nrf_desktop/configure/nrf52840_pca20041``.
+#. Optionally, depending on the board, edit the DTS overlay file.
+   This step is not required if you have created a new board and its DTS files fully describe your hardware.
+   In such case, the overlay file can be left empty.
+#. In Kconfig, ensure that the following modules that are specific for gaming mouse are enabled:
+
+    * :ref:`motion`
+    * :ref:`wheel`
+    * :ref:`buttons`
+    * :ref:`battery_meas`
+    * :ref:`leds`
+
+#. For each module enabled, change its ``_def`` file to match your hardware:
+
+    * Motion module
+        * The ``nrf52840_pca20041`` uses the PMW3360 optical motion sensor.
+          The sensor is configured in DTS, and the sensor type is selected in the application configuration.
+          If you need to add a new sensor, expand the application configuration.
+    * wheel module
+        * The wheel is based on the QDEC peripheral of the nRF52840 device and the hardware-related part is configured in DTS.
+    * buttons module
+        * To simplify the configuration of arrays, the nRF Desktop project uses ``_def`` files.
+        * The ``buttons`` module's ``_def`` file contains pins assigned to rows and columns.
+    * GATT Battery module
+        * The battery module ``_def`` file contains mapping needed to match the voltage read from ADC to the battery level.
+    * leds module
+        * The application uses two LEDs -- one for the peers state, and one for the system state indication.
+        * To control each LED brightness, the project uses the PWM channels.
+          The PWM peripheral needs to be configured in DTS files, and the leds module ``_def`` file needs to be configured to indicate which PWM channel is assigned to each LED color.
```

Comment:
Mention `led_state_def.h`?

Reply:
PWM has to be also enabled in Kconfig.

Reply:
led state def defines effects used for system state not how leds physical are linked to logical leds.

enabling PWM driver is worth mentioning

Reply:
done